### PR TITLE
fix(nuxt): ignore falsy modules

### PR DESCRIPTION
**Pull request by antfu**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This allows user to enable some modules conditionally, say:

```ts
export default defineNuxtConfig({
  modules: [
    '@vueuse/nuxt'
    process.env.CI ? undefined : '@nuxt/devtools',
  ],
})
```

Currently it will error out with:

```
[error] Nuxt module should be a function: undefined
```

The current workaround is:

```ts
export default defineNuxtConfig({
  modules: [
    '@vueuse/nuxt'
    ...(process.env.CI ? [] : ['@nuxt/devtools']),
  ],
})
```

Which I think it's not idle.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: test/fixtures/basic/nuxt.config.ts [hunk 1]
```diff
@@ -135,7 +135,9 @@ export default defineNuxtConfig({
     function (_, nuxt) {
       nuxt.options.optimization.treeShake.composables.server[nuxt.options.rootDir] = ['useClientOnlyComposable', 'setTitleToPink']
       nuxt.options.optimization.treeShake.composables.client[nuxt.options.rootDir] = ['useServerOnlyComposable']
-    }
+    },
+    // To test falsy module values
+    undefined,
```
Comment (danielroe) [inline test/fixtures/basic/nuxt.config.ts:140]:
```suggestion
    undefined
```

File: packages/schema/src/config/common.ts [hunk 2]
```diff
@@ -194,9 +194,11 @@ export default defineUntypedSchema({
    *   function () {}
    * ]
    * ```
-   * @type {(typeof import('../src/types/module').NuxtModule | string | [typeof import('../src/types/module').NuxtModule | string, Record<string, any>])[]}
+   * @type {(typeof import('../src/types/module').NuxtModule | string | [typeof import('../src/types/module').NuxtModule | string, Record<string, any>] | undefined | null | false)[]}
    */
-  modules: [],
+  modules: {
+    $resolve: val => [].concat(val).filter(Boolean)
```
Comment (antfu) [inline packages/schema/src/config/common.ts:200]: It can also come from layers etc, right?

Comment (danielroe) [reply packages/schema/src/config/common.ts:200]: I think this will handle layers too.

**Comment by danielroe** [comment]
It might also be worth considering https://github.com/unjs/c12/pull/61 as an alternative approach to this.

**Comment by Atinux** [comment]
They could actually do:

```ts
defineNuxtConfig({
  $development: {
     modules: ['@nuxt/devtools']
  },
  modules: ['@vueuse/nuxt']
})
```

In case `process.env.CI` is used during `nuxt build`

Also, you can also have:

```ts
defineNuxtConfig({
  modules: [
    '@vueuse/nuxt'
    '@nuxt/devtools',
  ],
  devtools: {
    enabled: !process.env.CI
  }
})
```

**Comment by pi0** [comment]
Today, modules can self-disable based on environment that also introduced environment specific `$development` / `$production`.

Still i see value in supporting such filtering because it covers custom logic such as CI+Target or so.

On related topic, we might thinking about `disabledModules: []`. Today it is impossible to easily disable a module provided by a nuxt layer.

**Review by danielroe — APPROVED** [review]

**Pull request merged**